app/models/manageiq/providers/infra_manager.rb

```diff
@@ -44,5 +44,24 @@ def self.merge_config_settings(cfg = VMDB::Config.new("vmdb"))
       path = [:ems, :ems_redhat, :service, :read_timeout]
       cfg.merge_from_template_if_missing(*path)
     end
+
+    def self.credential_validation(ems, auth_type, creds, revalidate)
+      ems.update_authentication(creds, :save => false)
+
+      ok, error_message = ems.authentication_check(auth_type, :save => revalidate)
+
+      raise MiqException::Error, error_message unless ok
+    end
+
+    def self.credential_validation_task(options)
```

Comment:
I think this (and the method above) should be moved to `ExtManagementSystem`.  Otherwise the call from `ems_common` controller will fail for all managers that are not Infra (e.g., OpenStack Cloud, Kubernetes, etc.).